services/community-site/src/utils/validators.ts

```diff
@@ -0,0 +1,34 @@
+import { Validator, ValidatorWithStats, YieldedValidator } from '../interfaces/Validator';
+
+const calulcateValidatorYield = (validators: Validator[]): ValidatorWithStats[] | Validator[] => {
+  const hasStats = (validators[0] as ValidatorWithStats)?.pbftsProduced !== 0;
+  if (!hasStats) {
+    return validators;
+  }
+  const validatorsWithStats = validators as ValidatorWithStats[];
+  const validatorsWithStake: YieldedValidator[] = validatorsWithStats.map((v) => {
+    return {
+      ...v,
+      blocksPerStake: v.pbftsProduced || 0 / Number.parseFloat(v.delegation.toString()),
+    };
+  });
+  const minBlockRatio = validatorsWithStake.reduce((prev, curr) =>
+    prev.blocksPerStake! < curr.blocksPerStake! ? prev : curr,
+  );
+
+  const maxBlockRatio = validatorsWithStake.reduce((prev, curr) =>
+    prev.blocksPerStake! > curr.blocksPerStake! ? prev : curr,
+  );
```

Comment:
This could lead to errors. I would use `Math.min(...validatorsWithStake.map(v => v.blocksPerStake))`